### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by rhatlapa**
Several useful improvements to config cloner, see commit messages for details.

### CODE REVIEW

**Comment by rhatlapa** [comment]
@olivergondza could you please review?

File: src/main/java/org/jenkinsci/tools/configcloner/Main.java [hunk 1]
```diff
@@ -47,7 +47,10 @@
     public static void main(final String[] args) {
 
         final CommandResponse resp = CommandResponse.system();
-        final CLIPool cliPool = new CLIPool(CLIFactory.system());
+        String sshKeyLocation = System.getenv("SSH_PRIVATE_KEY");
+        final CLIPool cliPool = new CLIPool(sshKeyLocation == null ?
+                CLIFactory.system() :
+                CLIFactory.provided(new String[]{sshKeyLocation}));
```
Comment (olivergondza) [inline src/main/java/org/jenkinsci/tools/configcloner/Main.java]: No need to wrap in `new String[]{...}`, the argument is variadic.

File: src/main/java/org/jenkinsci/tools/configcloner/UrlParser.java [hunk 2]
```diff
@@ -63,8 +73,8 @@ public final ConfigDestination destination(final String stringUrl) {
 
             final ConfigDestination dest = _parseDest(base, url);
 
-            if (dest.equals(base)) throw new IllegalArgumentException(
-                    "Source and destination is the same url"
+            if (dest.equals(base) && !ignoreSameJenkins) throw new IllegalArgumentException(
+                    "Source and destination is the same url and you are not forcing its override"
```
Comment (olivergondza) [inline src/main/java/org/jenkinsci/tools/configcloner/UrlParser.java]: There is probably a bug somewhere we need to fix instead. The cloner was never supposed to enforce the entities are from different Jenkinses - it makes sure you are not overwriting the same entity on the same Jenkins (think of it like `cp foo/bar foo/bar`). Can you please report what destinations are considered same, while they should not be? Thanks

File: src/main/java/org/jenkinsci/tools/configcloner/handler/TransferHandler.java [hunk 3]
```diff
@@ -132,7 +129,12 @@ private CommandResponse send(
                     destination, xmlString, this.updateCommandName(), destJob
             );
 
-            if (rsp.succeeded()) return response.returnCode(0);
+            if (rsp.succeeded()) {
+                return response.returnCode(0);
+            } else {
+                System.err.println("Updating xml failed with stderr:" + rsp.stderr());
+                System.out.println("Updating xml failed with stdout:" + rsp.stdout());
```
Comment (olivergondza) [inline src/main/java/org/jenkinsci/tools/configcloner/handler/TransferHandler.java]: As an error message, this should go to standard error (despite it was reported by underlying CLI command standard output). Also, for testability sake, do not print directly to `System.err/out`, use `response` instead.

**Review by olivergondza — CHANGES_REQUESTED** [review]

File: src/main/java/org/jenkinsci/tools/configcloner/UrlParser.java (hunk 2 above)
Comment (rhatlapa) [reply src/main/java/org/jenkinsci/tools/configcloner/UrlParser.java]: In my particular case I want to override exactly the same job as I am basically fixing some config of the job.

File: src/main/java/org/jenkinsci/tools/configcloner/handler/TransferHandler.java (hunk 3 above)
Comment (rhatlapa) [reply src/main/java/org/jenkinsci/tools/configcloner/handler/TransferHandler.java]: updated

File: src/main/java/org/jenkinsci/tools/configcloner/Main.java (hunk 1 above)
Comment (rhatlapa) [reply src/main/java/org/jenkinsci/tools/configcloner/Main.java]: Thanks, updated

**Comment by rhatlapa** [comment]
Thank you for review, I have done requested updates.

File: src/main/java/org/jenkinsci/tools/configcloner/UrlParser.java (hunk 2 above)
Comment (olivergondza) [reply src/main/java/org/jenkinsci/tools/configcloner/UrlParser.java]: Yeah, I guess with the `sed` functionality it makes sense to update it. What about permitting it with a warning, so accidents can be spotted?

Comment (rhatlapa) [reply src/main/java/org/jenkinsci/tools/configcloner/UrlParser.java]: Added warning message.

**Comment by rhatlapa** [comment]
Updated based on review, please review again.

**Comment by olivergondza** [comment]
Merged with slight modifications https://github.com/olivergondza/jenkins-config-cloner/commit/ae7be8d908068b705e5c55459ffb0ee035244f95. Thanks!

**Pull request merged**